Sources/MapboxNavigation/NavigationViewportDataSource.swift

```diff
@@ -174,106 +166,154 @@ public class NavigationViewportDataSource: ViewportDataSource {
                                routeProgress: RouteProgress? = nil) {
         guard let mapView = mapView else { return }
         
+        let followingCameraOptions = options.followingCameraOptions
+        
         if let location = rawLocation ?? passiveLocation {
-            let followingWithoutRouteZoomLevel = CGFloat(14.0)
+            if followingCameraOptions.centerUpdatesAllowed {
+                followingMobileCamera.center = location.coordinate
+                followingCarPlayCamera.center = location.coordinate
+            }
             
-            followingMobileCamera.center = location.coordinate
-            followingMobileCamera.zoom = followingWithoutRouteZoomLevel
-            followingMobileCamera.bearing = 0.0
-            followingMobileCamera.anchor = mapView.center
-            followingMobileCamera.pitch = 0.0
-            followingMobileCamera.padding = .zero
+            if followingCameraOptions.zoomUpdatesAllowed {
+                let altitude = 4000.0
+                let zoom = CGFloat(ZoomLevelForAltitude(altitude,
+                                                        mapView.pitch,
+                                                        location.coordinate.latitude,
+                                                        mapView.bounds.size))
+                
+                followingMobileCamera.zoom = zoom
+                followingCarPlayCamera.zoom = zoom
+            }
+            
+            if followingCameraOptions.bearingUpdatesAllowed {
+                followingMobileCamera.bearing = 0.0
+                followingCarPlayCamera.bearing = 0.0
+            }
             
-            followingCarPlayCamera.center = location.coordinate
-            followingCarPlayCamera.zoom = followingWithoutRouteZoomLevel
-            followingCarPlayCamera.bearing = 0.0
+            followingMobileCamera.anchor = mapView.center
             followingCarPlayCamera.anchor = mapView.center
-            followingCarPlayCamera.pitch = 0.0
-            followingCarPlayCamera.padding = .zero
+            
+            if followingCameraOptions.pitchUpdatesAllowed {
+                followingMobileCamera.pitch = 0.0
+                followingCarPlayCamera.pitch = 0.0
+            }
+            
+            if followingCameraOptions.paddingUpdatesAllowed {
+                followingMobileCamera.padding = .zero
+                followingCarPlayCamera.padding = .zero
+            }
             
             return
         }
         
         if let location = activeLocation, let routeProgress = routeProgress {
+            var compoundManeuvers: [[CLLocationCoordinate2D]] = []
+            let frameGeometryAfterManeuver = followingCameraOptions.frameGeometryAfterManeuver
             let pitchСoefficient = self.pitchСoefficient(routeProgress, currentCoordinate: location.coordinate)
             let pitch = maximumPitch * pitchСoefficient
```

Comment:
What about extracting `maximumPitch` to `options`? And what do you think about introducing `minimumPitch`(even with 0 as a default value)?

Reply:
`maximumPitch` is now exposed as `FollowingCameraOptions.defaultPitch`. Android doesn't include `minimumPitch` for now, so we'll keep it as it is (at least for now).